server/static/JSWebsocketClient.js

```diff
@@ -1,29 +1,131 @@
-// JS Websocket Client Sample
-// Author: Bhojan Anand, NUS SoC
-
-let audioContext;
-let socket = new WebSocket("ws://localhost:80/mic");
-
-// Capture microphone input
-navigator.mediaDevices.getUserMedia({ audio: true }).then(stream => {
-    let mediaRecorder = new MediaRecorder(stream);
-    mediaRecorder.ondataavailable = (event) => {
-        console.log('Audio data type:', event.data.type);  // Likely "audio/webm" or "audio/ogg"
-        socket.send(event.data);  // Send the audio data to the server
+// Initialize variables
+let audioContext = new (window.AudioContext || window.webkitAudioContext)();
+let mediaRecorder = null;
+let websocket = null;
+const statusDiv = document.getElementById('status');
+const recordButton = document.getElementById('recordButton');
+
+function setupAvatar() {
+    const name = decodeURIComponent(window.location.pathname.split('/').pop());
+    const avatarText = document.querySelector('.avatar-text');
+    if (avatarText) {
+        avatarText.setAttribute('data-letter', name.charAt(0).toUpperCase());
+    }
+}
+
+// Update status badge styles
+function updateStatus(message, type) {
+    const statusBadge = document.getElementById('status');
+    statusBadge.textContent = message;
+    
+    // Remove all status classes
+    statusBadge.classList.remove('connected', 'recording');
+    
+    // Add appropriate class
+    if (type) {
+        statusBadge.classList.add(type);
+    }
+}
+
+// Initialize WebSocket connection
+function initWebSocket() {
+    const name = window.location.pathname.split('/').pop();
+
+    websocket = new WebSocket('ws://localhost:8000/mic/' + name);
```

Comment:
should be `ws://localhost:8000/mic/` instead

Reply:
Fixed